src/model/test/ElectricLoadCenterStorageLiIonNMCBattery_GTest.cpp

```diff
@@ -0,0 +1,176 @@
+  EXPECT_TRUE(battery.setBatteryMass(1000));
+  EXPECT_TRUE(battery.setBatterySurfaceArea(2000));
+  EXPECT_TRUE(battery.setBatterySpecificHeatCapacity(3000));
+  EXPECT_TRUE(battery.setHeatTransferCoefficientBetweenBatteryandAmbient(6.5));
+  EXPECT_TRUE(battery.setFullyChargedCellVoltage(3.1));
+  EXPECT_TRUE(battery.setCellVoltageatEndofExponentialZone(3.25));
+  EXPECT_TRUE(battery.setCellVoltageatEndofNominalZone(3.35));
+  EXPECT_TRUE(battery.setDefaultNominalCellVoltage(3.15));
+  EXPECT_TRUE(battery.setFullyChargedCellCapacity(3.12));
+  EXPECT_TRUE(battery.setFractionofCellCapacityRemovedattheEndofExponentialZone(0.73));
+  EXPECT_TRUE(battery.setFractionofCellCapacityRemovedattheEndofNominalZone(0.74));
+  EXPECT_TRUE(battery.setChargeRateatWhichVoltagevsCapacityCurveWasGenerated(4000));
+  EXPECT_TRUE(battery.setBatteryCellInternalElectricalResistance(6000));
+
+  Schedule schedule = battery.availabilitySchedule();
+  boost::optional<ScheduleConstant> scheduleConstant = schedule.optionalCast<ScheduleConstant>();
+  ASSERT_TRUE(scheduleConstant);
+  EXPECT_EQ((*scheduleConstant).value(), 0.5);
+  EXPECT_TRUE(battery.thermalZone());
+  EXPECT_EQ(0.5, battery.radiativeFraction());
+  EXPECT_EQ("None", battery.lifetimeModel());
+  EXPECT_EQ(5, battery.numberofCellsinSeries());
+  EXPECT_EQ(100, battery.numberofStringsinParallel());
+  EXPECT_EQ(0.51, battery.initialFractionalStateofCharge());
+  EXPECT_EQ(0.52, battery.dctoDCChargingEfficiency());
+  EXPECT_EQ(1000, battery.batteryMass());
+  EXPECT_EQ(2000, battery.batterySurfaceArea());
+  EXPECT_EQ(3000, battery.batterySpecificHeatCapacity());
+  EXPECT_EQ(6.5, battery.heatTransferCoefficientBetweenBatteryandAmbient());
+  EXPECT_EQ(3.1, battery.fullyChargedCellVoltage());
+  EXPECT_EQ(3.25, battery.cellVoltageatEndofExponentialZone());
+  EXPECT_EQ(3.35, battery.cellVoltageatEndofNominalZone());
+  EXPECT_EQ(3.15, battery.defaultNominalCellVoltage());
+  EXPECT_EQ(3.12, battery.fullyChargedCellCapacity());
+  EXPECT_EQ(0.73, battery.fractionofCellCapacityRemovedattheEndofExponentialZone());
+  EXPECT_EQ(0.74, battery.fractionofCellCapacityRemovedattheEndofNominalZone());
+  EXPECT_EQ(4000, battery.chargeRateatWhichVoltagevsCapacityCurveWasGenerated());
+  EXPECT_EQ(6000, battery.batteryCellInternalElectricalResistance());
+
+  battery.resetThermalZone();
+
+  EXPECT_FALSE(battery.thermalZone());
+}
+
+TEST_F(ModelFixture, ElectricLoadCenterStorageLiIonNMCBattery_Clone) {
+  Model m;
+  ThermalZone z(m);
+  ElectricLoadCenterStorageLiIonNMCBattery battery(m);
+  EXPECT_TRUE(battery.setThermalZone(z));
+
+  battery.setRadiativeFraction(0.6);
+
+  ElectricLoadCenterStorageLiIonNMCBattery batteryClone = battery.clone(m).cast<ElectricLoadCenterStorageLiIonNMCBattery>();
+  EXPECT_EQ(0.6, batteryClone.radiativeFraction());
+  // TODO: should the clone retain the thermalZone or clear it?
+  EXPECT_TRUE(batteryClone.thermalZone());
+  EXPECT_EQ(z, batteryClone.thermalZone());
```

Comment:
@kbenne @joseph-robertson This is the last TODO IMHO

Reply:
I noticed that ElectricLoadCenterStorageConverter for eg keeps it, so I'm fine leave it as is